cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -49,33 +50,35 @@ type azureCache struct {
 	refreshInterval time.Duration
 
 	// Cache content.
-	resourceGroup        string
-	vmType               string
-	vmsPoolSet           map[string]struct{} // track the nodepools that're vms pool
-	scaleSets            map[string]compute.VirtualMachineScaleSet
-	virtualMachines      map[string][]compute.VirtualMachine
-	registeredNodeGroups []cloudprovider.NodeGroup
-	instanceToNodeGroup  map[azureRef]cloudprovider.NodeGroup
-	unownedInstances     map[azureRef]bool
-	autoscalingOptions   map[azureRef]map[string]string
-	skus                 map[string]*skewer.Cache
+	resourceGroup          string
+	vmType                 string
+	vmsPoolSet             map[string]struct{} // track the nodepools that're vms pool
+	scaleSets              map[string]compute.VirtualMachineScaleSet
+	virtualMachines        map[string][]compute.VirtualMachine
+	registeredNodeGroups   []cloudprovider.NodeGroup
+	registeredAKSPoolNames sets.Set[string]
+	instanceToNodeGroup    map[azureRef]cloudprovider.NodeGroup
+	unownedInstances       map[azureRef]bool
+	autoscalingOptions     map[azureRef]map[string]string
+	skus                   map[string]*skewer.Cache
 }
 
 func newAzureCache(client *azClient, cacheTTL time.Duration, resourceGroup, vmType string, enableDynamicInstanceList bool, defaultLocation string) (*azureCache, error) {
 	cache := &azureCache{
-		interrupt:            make(chan struct{}),
-		azClient:             client,
-		refreshInterval:      cacheTTL,
-		resourceGroup:        resourceGroup,
-		vmType:               vmType,
-		vmsPoolSet:           make(map[string]struct{}),
-		scaleSets:            make(map[string]compute.VirtualMachineScaleSet),
-		virtualMachines:      make(map[string][]compute.VirtualMachine),
-		registeredNodeGroups: make([]cloudprovider.NodeGroup, 0),
-		instanceToNodeGroup:  make(map[azureRef]cloudprovider.NodeGroup),
-		unownedInstances:     make(map[azureRef]bool),
-		autoscalingOptions:   make(map[azureRef]map[string]string),
-		skus:                 make(map[string]*skewer.Cache),
+		interrupt:              make(chan struct{}),
+		azClient:               client,
+		refreshInterval:        cacheTTL,
+		resourceGroup:          resourceGroup,
+		vmType:                 vmType,
+		vmsPoolSet:             make(map[string]struct{}),
+		scaleSets:              make(map[string]compute.VirtualMachineScaleSet),
+		virtualMachines:        make(map[string][]compute.VirtualMachine),
+		registeredNodeGroups:   make([]cloudprovider.NodeGroup, 0),
+		registeredAKSPoolNames: make(sets.Set[string]),
```

Comment:
Is this just to ease up the process of getting matched `strings.ToLower(registeredNodeGroup.Id())`? How much does it help, and is it really worth it?

Reply:
providerID normalization seems very worth it in this case. Some apis return it in different casing so lowercasing it seems logical to me.